server/plugin/hooks.go

```diff
@@ -21,18 +21,25 @@ func (p *Plugin) MessageHasBeenPosted(c *plugin.Context, post *model.Post) {
 		return
 	}
 
-	channel, appErr := p.API.GetChannel(post.ChannelId)
-	if appErr != nil {
-		p.API.LogError("Error occurred while fetching channel by ID. ChannelID: %s. Error: %s", post.ChannelId, appErr.Error())
-		return
-	}
+	var channelName string
+	cacheVal, err := p.dmChannelCache.Get(post.ChannelId)
+	if err == nil {
+		channelName = cacheVal.(string)
+	} else {
+		channel, channelErr := p.API.GetChannel(post.ChannelId)
+		if channelErr != nil {
+			p.API.LogError("Error occurred while fetching the channel by ID. ChannelID: %s. Error: %s", post.ChannelId, channelErr.Error())
+			return
+		}
 
-	if channel.Type != model.CHANNEL_DIRECT {
-		return
+		channelName = channel.Name
+		if err = p.dmChannelCache.SetWithExpire(post.ChannelId, channelName, time.Minute*time.Duration(p.getConfiguration().ChannelCacheTTL)); err != nil {
```

Comment:
What is the advantage of storing the channel name vs. storing a boolean flag that indicates if a channel is a direct one with the bot?